In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
url='https://raw.githubusercontent.com/helen-oy/HamoyeFOOD-dataset/main/FoodHamoye.csv'

In [3]:
df=pd.read_csv(url, encoding = "ISO-8859-1")

In [4]:
df.describe(include = 'all')

Area Code   Area     Item Code                     Item  \
count   60943.000000  60943  60943.000000                    60943   
unique           NaN     49           NaN                      119   
top              NaN  Kenya           NaN  Milk - Excluding Butter   
freq             NaN   1560           NaN                     1262   
mean      134.265576    NaN   2687.176706                      NaN   
std        72.605709    NaN    146.055739                      NaN   
min         4.000000    NaN   2501.000000                      NaN   
25%        74.000000    NaN   2562.000000                      NaN   
50%       136.000000    NaN   2630.000000                      NaN   
75%       195.000000    NaN   2775.000000                      NaN   
max       276.000000    NaN   2961.000000                      NaN   

        Element Code                   Element         Unit          Y2014  \
count   60943.000000                     60943        60943   59354.000000   
unique           NaN                        18            5            NaN   
top              NaN  Domestic supply quantity  1000 tonnes            NaN   
freq             NaN                      5295        40933            NaN   
mean     3814.856456                       NaN          NaN     134.196282   
std      2212.007033                       NaN          NaN    1567.663696   
min       511.000000                       NaN          NaN   -1796.000000   
25%       684.000000                       NaN          NaN       0.000000   
50%      5142.000000                       NaN          NaN       0.090000   
75%      5511.000000                       NaN          NaN       8.340000   
max      5911.000000                       NaN          NaN  176405.000000   

                Y2015          Y2016          Y2017          Y2018  
count    59395.000000   59408.000000   59437.000000   59507.000000  
unique            NaN            NaN            NaN            NaN  
top               NaN            NaN            NaN            NaN  
freq              NaN            NaN            NaN            NaN  
mean       135.235966     136.555222     140.917765     143.758381  
std       1603.403984    1640.007194    1671.862359    1710.782658  
min      -3161.000000   -3225.000000   -1582.000000   -3396.000000  
25%          0.000000       0.000000       0.000000       0.000000  
50%          0.080000       0.080000       0.100000       0.070000  
75%          8.460000       8.430000       9.000000       9.000000  
max     181137.000000  185960.000000  190873.000000  195875.000000

# Q20 - The total number of unique countries is 49 from the df.describe(include='all') output unique area

In [5]:
df.groupby('Item')['Y2014','Y2015', 'Y2016', 'Y2017'].sum()

C:\Users\HP\AppData\Local\Temp/ipykernel_8828/231100273.py:1: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  df.groupby('Item')['Y2014','Y2015', 'Y2016', 'Y2017'].sum()


Y2014      Y2015      Y2016      Y2017
Item                                                           
Alcohol, Non-Food      2403.00    2180.00    2223.00    2348.00
Alcoholic Beverages  102410.11   98783.72   96958.75   95581.06
Animal Products       11935.65   11811.73   11661.69   11547.65
Animal fats          209460.54  200675.72  183314.22  269617.53
Apples and products    9499.23   10559.15    9853.26   10198.90
...                        ...        ...        ...        ...
Vegetables, Other    155038.96  158104.08  156218.90  157752.59
Vegetal Products     107145.19  107064.17  106792.72  107655.20
Wheat and products   232670.13  234710.51  239377.07  240047.62
Wine                   4497.36    4251.81    3872.09    4178.02
Yams                 200396.96  203151.78  220626.20  229174.59

[119 rows x 4 columns]

# Q11 - Answer is C 
the total sum of Animal Fat produced in 2014 and 2017 respectively is 209460.54 and 269617.53

In [6]:
round(df['Y2015'].describe(), 3)

count     59395.000
mean        135.236
std        1603.404
min       -3161.000
25%           0.000
50%           0.080
75%           8.460
max      181137.000
Name: Y2015, dtype: float64

# Q12- Answer is E
the mean is 135.236 and the standard deviation (std) is 1603.404

In [7]:
df.isnull().sum()

Area Code          0
Area               0
Item Code          0
Item               0
Element Code       0
Element            0
Unit               0
Y2014           1589
Y2015           1548
Y2016           1535
Y2017           1506
Y2018           1436
dtype: int64

In [8]:
percentage_missing_data = df. isnull(). sum() * 100/len(df)


In [9]:
round(percentage_missing_data, 2)

Area Code       0.00
Area            0.00
Item Code       0.00
Item            0.00
Element Code    0.00
Element         0.00
Unit            0.00
Y2014           2.61
Y2015           2.54
Y2016           2.52
Y2017           2.47
Y2018           2.36
dtype: float64

# Q13 - Answer is D
The total number and percentage of missing data in 2016 to 2 decimal places is 1535 and 2.52 respectively  

In [10]:
df.groupby('Element Code')['Y2014','Y2015', 'Y2016', 'Y2017'].corr()

C:\Users\HP\AppData\Local\Temp/ipykernel_8828/3382489259.py:1: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  df.groupby('Element Code')['Y2014','Y2015', 'Y2016', 'Y2017'].corr()


Y2014     Y2015     Y2016     Y2017
Element Code                                              
511          Y2014  1.000000  0.999988  0.999951  0.999885
             Y2015  0.999988  1.000000  0.999987  0.999947
             Y2016  0.999951  0.999987  1.000000  0.999986
             Y2017  0.999885  0.999947  0.999986  1.000000
645          Y2014  1.000000  0.997805  0.995394  0.994318
...                      ...       ...       ...       ...
5611         Y2017  0.987241  0.986712  0.978215  1.000000
5911         Y2014  1.000000  0.976942  0.992694  0.941934
             Y2015  0.976942  1.000000  0.965195  0.863665
             Y2016  0.992694  0.965195  1.000000  0.963790
             Y2017  0.941934  0.863665  0.963790  1.000000

[72 rows x 4 columns]

In [11]:
max(df.groupby('Element Code')['Y2014','Y2015', 'Y2016', 'Y2017'].corr())

C:\Users\HP\AppData\Local\Temp/ipykernel_8828/2361028243.py:1: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  max(df.groupby('Element Code')['Y2014','Y2015', 'Y2016', 'Y2017'].corr())


'Y2017'

# Q14 - Answer is D
the highest correlation with 'Element Code' is in 2017

In [12]:
df.groupby('Element')['Y2014','Y2015', 'Y2016', 'Y2017'].sum()

C:\Users\HP\AppData\Local\Temp/ipykernel_8828/42112715.py:1: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  df.groupby('Element')['Y2014','Y2015', 'Y2016', 'Y2017'].sum()


Y2014       Y2015       Y2016  \
Element                                                                      
Domestic supply quantity                1996716.35  2021493.55  2044842.70   
Export Quantity                          150020.64   157614.47   151920.46   
Fat supply quantity (g/capita/day)        10225.56    10235.74    10102.77   
Feed                                     216927.89   225050.22   228958.65   
Food                                    1212332.49  1232361.10  1247022.17   
Food supply (kcal/capita/day)            454257.00   453383.00   451810.00   
Food supply quantity (kg/capita/yr)       49650.63    49345.13    48985.28   
Import Quantity                          274144.48   267018.46   286582.78   
Losses                                   153223.00   155439.00   157787.00   
Other uses (non-food)                     78718.13    66254.41    69563.68   
Processing                               282923.00   287929.00   280631.00   
Production                              1931287.75  1947019.39  1943537.15   
Protein supply quantity (g/capita/day)    11836.46    11833.95    11779.69   
Residuals                                 30149.00    30045.00    37224.00   
Seed                                      21922.92    23976.82    23389.20   
Stock Variation                           58749.83    34910.99    33140.12   
Total Population - Both sexes           1031585.00  1058081.00  1085107.00   
Tourist consumption                         416.00      349.00       89.00   

                                             Y2017  
Element                                             
Domestic supply quantity                2088198.10  
Export Quantity                          182338.80  
Fat supply quantity (g/capita/day)        10253.84  
Feed                                     223705.68  
Food                                    1258888.28  
Food supply (kcal/capita/day)            454681.00  
Food supply quantity (kg/capita/yr)       48690.04  
Import Quantity                          294559.09  
Losses                                   160614.00  
Other uses (non-food)                     91645.97  
Processing                               292836.00  
Production                              2030056.89  
Protein supply quantity (g/capita/day)    11842.45  
Residuals                                 35500.00  
Seed                                      24870.14  
Stock Variation                           54316.91  
Total Population - Both sexes           1112641.00  
Tourist consumption                          91.00

# Q15 - Answer is D 
The highest sum of import quantity is 294559.09 in the year 2017

In [13]:
df.groupby('Element')['Y2014'].sum()

Element
Domestic supply quantity                  1996716.35
Export Quantity                            150020.64
Fat supply quantity (g/capita/day)          10225.56
Feed                                       216927.89
Food                                      1212332.49
Food supply (kcal/capita/day)              454257.00
Food supply quantity (kg/capita/yr)         49650.63
Import Quantity                            274144.48
Losses                                     153223.00
Other uses (non-food)                       78718.13
Processing                                 282923.00
Production                                1931287.75
Protein supply quantity (g/capita/day)      11836.46
Residuals                                   30149.00
Seed                                        21922.92
Stock Variation                             58749.83
Total Population - Both sexes             1031585.00
Tourist consumption                           416.00
Name: Y2014, dtype: float64

# Q16 - Answer is E
the total number ofsum of production is  1931287.75 in  2014

In [14]:
df.groupby('Element')['Y2018'].sum()
df.groupby('Element')['Y2018'].sum().sort_values(ascending=False)

Element
Domestic supply quantity                  2161192.10
Production                                2075072.89
Food                                      1303841.28
Total Population - Both sexes             1140605.00
Food supply (kcal/capita/day)              455261.00
Processing                                 308429.00
Import Quantity                            287997.09
Feed                                       233489.68
Export Quantity                            181594.80
Losses                                     163902.00
Other uses (non-food)                       91300.97
Food supply quantity (kg/capita/yr)         49056.85
Residuals                                   34864.00
Seed                                        25263.14
Stock Variation                             20577.91
Protein supply quantity (g/capita/day)      11833.56
Fat supply quantity (g/capita/day)          10258.69
Tourist consumption                            90.00
Name: Y2018, dtype: float64

In [15]:
max(df.groupby(['Element'])['Y2018'].sum())

2161192.1

# Q17 -Answer is C 
 the 'Element' that had the highest sum is domestic supply quantity

In [16]:
df.groupby('Element')['Y2018'].sum().sort_values(ascending=True)

Element
Tourist consumption                            90.00
Fat supply quantity (g/capita/day)          10258.69
Protein supply quantity (g/capita/day)      11833.56
Stock Variation                             20577.91
Seed                                        25263.14
Residuals                                   34864.00
Food supply quantity (kg/capita/yr)         49056.85
Other uses (non-food)                       91300.97
Losses                                     163902.00
Export Quantity                            181594.80
Feed                                       233489.68
Import Quantity                            287997.09
Processing                                 308429.00
Food supply (kcal/capita/day)              455261.00
Total Population - Both sexes             1140605.00
Food                                      1303841.28
Production                                2075072.89
Domestic supply quantity                  2161192.10
Name: Y2018, dtype: float64

# Q18 - Answer is A
the third lowest sum in 2018 is Protein supply quantity(g/capita/day)

In [17]:
df.groupby(['Area', 'Element'])['Y2018'].sum()

Area      Element                               
Algeria   Domestic supply quantity                  90023.42
          Export Quantity                             811.54
          Fat supply quantity (g/capita/day)          309.09
          Feed                                      10668.00
          Food                                      63736.33
                                                      ...   
Zimbabwe  Protein supply quantity (g/capita/day)      176.57
          Residuals                                  -546.00
          Seed                                        140.00
          Stock Variation                            -447.00
          Total Population - Both sexes             14439.00
Name: Y2018, Length: 803, dtype: float64

In [18]:
tot_import_quantity = df.loc[(df['Element']== 'Import Quantity') & (df['Area']== 'Algeria')]['Y2018'].sum()
print(tot_import_quantity)

36238.29


# Q19- Answer is C 
The total import quantity for the year 2018 is 36238.29

In [19]:
df.describe(include='all')

Area Code   Area     Item Code                     Item  \
count   60943.000000  60943  60943.000000                    60943   
unique           NaN     49           NaN                      119   
top              NaN  Kenya           NaN  Milk - Excluding Butter   
freq             NaN   1560           NaN                     1262   
mean      134.265576    NaN   2687.176706                      NaN   
std        72.605709    NaN    146.055739                      NaN   
min         4.000000    NaN   2501.000000                      NaN   
25%        74.000000    NaN   2562.000000                      NaN   
50%       136.000000    NaN   2630.000000                      NaN   
75%       195.000000    NaN   2775.000000                      NaN   
max       276.000000    NaN   2961.000000                      NaN   

        Element Code                   Element         Unit          Y2014  \
count   60943.000000                     60943        60943   59354.000000   
unique           NaN                        18            5            NaN   
top              NaN  Domestic supply quantity  1000 tonnes            NaN   
freq             NaN                      5295        40933            NaN   
mean     3814.856456                       NaN          NaN     134.196282   
std      2212.007033                       NaN          NaN    1567.663696   
min       511.000000                       NaN          NaN   -1796.000000   
25%       684.000000                       NaN          NaN       0.000000   
50%      5142.000000                       NaN          NaN       0.090000   
75%      5511.000000                       NaN          NaN       8.340000   
max      5911.000000                       NaN          NaN  176405.000000   

                Y2015          Y2016          Y2017          Y2018  
count    59395.000000   59408.000000   59437.000000   59507.000000  
unique            NaN            NaN            NaN            NaN  
top               NaN            NaN            NaN            NaN  
freq              NaN            NaN            NaN            NaN  
mean       135.235966     136.555222     140.917765     143.758381  
std       1603.403984    1640.007194    1671.862359    1710.782658  
min      -3161.000000   -3225.000000   -1582.000000   -3396.000000  
25%          0.000000       0.000000       0.000000       0.000000  
50%          0.080000       0.080000       0.100000       0.070000  
75%          8.460000       8.430000       9.000000       9.000000  
max     181137.000000  185960.000000  190873.000000  195875.000000

In [20]:
df1 = df.describe(include='all')

In [21]:
df1.loc['unique']['Area']

49

# Q20 - Answer is B
Column 'Area' represents countries, the total number of unique countries is 49 from the df.describe(include='all') output of (unique, area)